src/librustc_mir/transform/check_unsafety.rs

```diff
@@ -624,40 +648,61 @@ pub fn check_unsafety(tcx: TyCtxt<'_>, def_id: DefId) {
     let UnsafetyCheckResult { violations, unsafe_blocks } =
         tcx.unsafety_check_result(def_id.expect_local());
 
-    for &UnsafetyViolation { source_info, description, details, kind } in violations.iter() {
+    for &UnsafetyViolation { source_info, lint_root, description, details, kind } in
+        violations.iter()
+    {
         // Report an error.
+        let unsafe_fn_msg =
+            if unsafe_op_in_unsafe_fn_allowed(tcx, lint_root) { "" } else { " function or" };
```

Comment:
Well I think I just... err... flipped the condition here... I don't know why, this is the second time I make a stupid mistake here -_-

Reply:
I was going to say...this looks backwards...